node/src/cli.rs

```diff
@@ -1,23 +1,46 @@
 use sc_cli::RunCmd;
+use std::collections::HashMap;
 use structopt::StructOpt;
 
 #[derive(Debug, StructOpt)]
 pub struct GatewayCmd {
-    /// Set the ETH Key ID for AWS KMS integration
-    #[structopt(long = "eth-key-id")]
-    pub eth_key_id: Option<String>,
+    // collect args after -- like cargo, eg cargo run -- --help for generic key-value configuration
+    // expected format is --eth-key-id abcdefg --miner 123456789
+    // OR
+    // ETH_KEY_ID abcdefg MINER 123456789
+    // equals = linking not supported as in --eth-key-id=abcdefg
+    #[structopt()]
+    pub args: Vec<String>,
+}
+
+impl GatewayCmd {
```

Comment:
Hmm can we just use a StructOpt with all of the ones we accept, and then convert that into a dynamic HashMap, so we can see what options are currently supported and avoid this parsing logic?

Reply:
The whole issue of `parse` is so that we can have env vars the way we expect them to be and CLI args the way we expect them to be - eg `--eth-rpc-url` on the cli and `ETH_RPC_URL` in the environment. This allows us to do overrides in a very consistent way while providing some level of backward compatibility.

We could use structopt for the ones we already accept but i thought it would be more confusing to have to add _only the new CLI params added after this pr_ after `--` the whole issue of the `--` thing is confusing enough on its own without having to say `--eth-rpc-url abcdef -- --flow-rpc-url ghijkl` its really easy to mess that up.

Reply:
I guess I'm confused...what are the old ones?

Reply:
There are currently 4 cli params that we add for gateway, those, in my scenario, are the "old ones". What i'm saying is that if we keep those 4 the way they are then when we require a "new" one, eg `--flow-rpc-url` we will need to put it _after_ the `--` so it will look like this 

```
#                    vvvvv existing params vvvvvv          vvvv all "new" params would have to go after the --
./gateway ... --eth-rpc-url ...                            --   --flow-rpc-url ... 
```

the way that I implemented it is 

```
#               vvvvv substrate params vvvvv           vvvv   all gateway params vvvvv
./gateway ... --unsafe-rpc ...                       --      --eth-rpc-url http://...  --flow-rpc-url ... 
```

hopefully this illustration clears up this question for you

Reply:
It makes sense to me that all our custom Gateway params come after the `--`, but how does that break using a StructOpt to describe what they are (and converting that to a HashMap)?

Reply:
The short answer to your question is that the two are mutually exclusive. The literal function of the `--` is "throw the rest into the last `Vec<String>` bag. Hopefully this makes sense?

Reply:
I guess I was thinking we could just take the throw away bag and use StructOpt to parse that separately, but maybe thats now how it works. Anyway sounds like `-e` will be the preferred method anyway and docs have list of accepted params (although I am sure those will get out of date easily)